ldm/dream/console_application.py

```diff
@@ -0,0 +1,303 @@
+            
+            # TODO: probably need to have a file writing utility
+            # TODO: Grids should be generated as part of the generator (or future pipelining)
+            if opt.grid and len(job.results) > 0:
+                grid_img   = make_grid(list(map(get_image, job.results)))
+                grid_seeds = list(map(lambda dreamResult: dreamResult.id, job.results))
+                first_seed = job.results[0].seed
+                filename   = f'{job.id}.{first_seed}.grid.png'
+                formatted_dream_prompt  = opt.dream_prompt_str(seed=first_seed,grid=True,iterations=len(job.results))
+                formatted_dream_prompt += f' # {grid_seeds}'
+                metadata = metadata_dumps( # TODO: standardize metadata with model class
+                    opt,
+                    seeds      = grid_seeds,
+                    model_hash = generator.model_hash
+                    )
+                path = file_writer.save_image_and_prompt_to_png(
+                    image        = grid_img,
+                    dream_prompt = formatted_dream_prompt, # NOTE: metadata already has all this information
+                    metadata     = json.dumps(metadata),
+                    name         = filename
+                )
+
+                # TODO: Make sure opt.save_original cleans up non-final results here
+                results = [[path, formatted_dream_prompt]]
+            else:
+                # TODO: Make sure opt.save_original cleans up non-final results here
+                results = list(map(lambda dreamResult: [
+                  image_storage_service.path(dreamResult.id),
+                  opt.dream_prompt_str(seed=dreamResult.seed)
+                ], job.results))
+
+        except AssertionError as e:
+            print(e)
+            continue
+
+        except OSError as e:
+            print(e)
+            continue
+
+        print('Outputs:')
+        log_path = os.path.join(current_outdir, 'dream_log.txt')
+        write_log_message(results, log_path)
+        print()
+
+    print('goodbye!')
+
+
+def get_next_command(infile=None) -> str:  # command string
+    if infile is None:
+        command = input('dream> ')
+    else:
+        command = infile.readline()
+        if not command:
+            raise EOFError
+        else:
+            command = command.strip()
+        if len(command)>0:
+            print(f'#{command}')
+    return command
+
+output_cntr = 0
+def write_log_message(results, log_path):
+    """logs the name of the output image, prompt, and prompt args to the terminal and log file"""
+    global output_cntr
+    log_lines = [f'{path}: {prompt}\n' for path, prompt in results]
+    for l in log_lines:
+        output_cntr += 1
+        print(f'[{output_cntr}] {l}',end='')
+
+
+    with open(log_path, 'a', encoding='utf-8') as file:
+        file.writelines(log_lines)
+
+
+if __name__ == '__main__':
+  """Load configuration and run application"""
```

Comment:
Seems like here is 2 spaces, and in other parts there are 4 spaces. Best keep it the same all over.
Also, super highly suggest moving to use a formatter. `black` has become a standard and would fix all these kinds of issues.

Reply:
Project formatting standards are incoming see https://github.com/lstein/stable-diffusion/discussions/631